src/chain.h

```diff
@@ -341,6 +348,8 @@ class CDiskBlockIndex : public CBlockIndex
             READWRITE(VARINT(nDataPos));
         if (nStatus & BLOCK_HAVE_UNDO)
             READWRITE(VARINT(nUndoPos));
+        if (nStatus & BLOCK_OPT_ACTIVATES_NU)
+            READWRITE(nBranchId);
```

Comment:
Mostly because we don't *need* to use any disk space to store it. We can rely on the invariant that blocks not changing the consensus rules will always have the same consensus branch ID as their parent, and set up the in-memory representation on startup (like is done for the `nChain*` values). I concede it's not a strong argument though, given that it would only be an additional 1MiB for the current blockchain height (which is about 0.1% overhead on a fully-synced node with `txindex=1`; it would be a larger percentage without that, but I don't have a non-txindex node to compare it to).

The only reason I'm even storing it at all is to be able to efficiently detect when the indexed blocks don't match the expected epoch chain. Figuring that out can only be done by validating each block against all the consensus rules (hence a validity level of at least `SCRIPTS`), which is expensive to do on start. If we could just assume the block index was valid, or didn't need to care about that, then we could just always use the height -> branch ID mapping functions.

Reply:
Okay, this seems fine then.